### ISSUE TITLE
Build for arm64 linux

### ISSUE THREAD

**Issue by JoshAtticus** [Mastermindzh/tidal-hifi#311]
Would be nice to be able to run on arm64 linux without having to compile myself

**Comment by Spl0itable** [comment]
Any movement on this? Thanks!

**Comment by andirsun** [comment]
I am opening a new update inside electron by castlabs with a solution for Widevine on arm64 platforms like Fedora Asahi for Apple M series macbooks https://github.com/castlabs/electron-releases/issues/23#issuecomment-1874133501

**Comment by Mastermindzh** [comment]
Hey,

Yeah still waiting for Castlab to introduce ARM support eventually.
I mean, we can make an ARM build available if you want but it won't have widevine support I reckon.

I don't know if that's a problem since you're now building it yourself anyway.
Let me know.

**Comment by JoshAtticus** [comment]
> Hey,
> 
> Yeah still waiting for Castlab to introduce ARM support eventually. I mean, we can make an ARM build available if you want but it won't have widevine support I reckon.
> 
> I don't know if that's a problem since you're now building it yourself anyway. Let me know.

I initially made my issue related to M1 Macs with Asahi Linux. Asahi uses https://github.com/AsahiLinux/widevine-installer for Widevine and it *might* be possible to implement that and maybe even make it work on Raspberry Pis

**Comment by andirsun** [comment]
@Mastermindzh We could uses the asahi linux widevine installer for arm64 builds. I can test it if you want.

**Comment by Mastermindzh** [comment]
Sure, give it a whirl and let me know. I expect to work on a new version of Tidal Friday so we might be able to include this in the weekend :)

**Comment by Spl0itable** [comment]
> Hey,
> 
> Yeah still waiting for Castlab to introduce ARM support eventually. I mean, we can make an ARM build available if you want but it won't have widevine support I reckon.
> 
> I don't know if that's a problem since you're now building it yourself anyway. Let me know.

Yes, I am using an ubuntu vm on m1 mac. I'd be happy to test without widevine support and see if there's any issues without the drm. Thanks!

**Comment by andirsun** [comment]
Well the app build properly but I am trying to add widevine support on Fedora for Macbooks.
I have support for widevine on Firefox and Chromium browsers as you can see here
![Screenshot from 2024-01-04 22-54-15](https://github.com/Mastermindzh/tidal-hifi/assets/22928302/0e128874-56e4-4282-a4c1-3892b2851681)

The cdm is here
![Screenshot from 2024-01-04 23-00-32](https://github.com/Mastermindzh/tidal-hifi/assets/22928302/8770c178-8b85-49b9-b816-7995a480577d)

But after trying to make electron read the Widevine from path as recomended by Fedora mantainers 
![Screenshot from 2024-01-04 23-03-39](https://github.com/Mastermindzh/tidal-hifi/assets/22928302/97432a9b-00b6-411c-b7ca-4e36dd1984b1)

```
// * `widevinecdm.dll` on Windows.
try {
  app.commandLine.appendSwitch('widevine-cdm-path', '/usr/lib64/chromium-browser/WidevineCdm/_platform_specific/linux_arm64/libwidevinecdm.so')
  app.commandLine.appendSwitch('widevine-cdm-version', '4.10.2662.3')
  console.log(`Widevine CDM path: ${app.commandLine.getSwitchValue('widevine-cdm-path')}`)
} catch (error) {
  console.log(error)
}
```

Widevine is not being taken from proper directory.
![Screenshot from 2024-01-04 22-55-03](https://github.com/Mastermindzh/tidal-hifi/assets/22928302/2b280328-524f-4afe-b803-50e866a9d72d)

I feel I am close to fix it but dont know how.

**Comment by Spl0itable** [comment]
@andirsun how did this go? Were you able to get an arm64 build working?

**Comment by andirsun** [comment]
@Spl0itable Actually no, I was not able to make electron read local Widevine CDM, so I ended creating a PWA shorcut using Chromium.
I think our best shot is to wait for castlabs to support arm64 officially.
![Screenshot from 2024-01-10 12-55-20](https://github.com/Mastermindzh/tidal-hifi/assets/22928302/1ba7162e-5d72-4de5-8e49-fa9cf644b6b5)

**Comment by Spl0itable** [comment]
@andirsun Ah no, sorry to hear that. Wish it had worked. Thanks for trying.

**Comment by Mastermindzh** [comment]
Thanks for all the effort @andirsun.
Let's hope the ARM support issue/ticket is prioritized relatively quickly :)

**Comment by cenunix** [comment]
so is this possible to use if I compile myself? or will i still be limited without widevine?

**Comment by andirsun** [comment]
Yep you can compile the project by yourself but unless you are able to make electron read you widevine path to reproduce DRM protected content, the app will not play anything.
